tools/chef/roles/db-mysql-upgrade.json

```diff
@@ -0,0 +1,62 @@
+{
+  "name": "db-mysql-upgrade",
+  "chef_type": "role",
+  "json_class": "Chef::Role",
+  "description": "MySQL upgrader. Be careful!",
+  "default_attributes": {
+    "mysql-libs": {
+      "replace_packages": [
+        "mysql-libs",
+        "mysql55w-libs",
+        "libmysqlclient16",
+        "mariadb-libs",
+        "Percona-Server-shared-55"
+      ]
+    },
+    "mysql-client": {
+      "replace_packages": [
+        "mysql",
+        "mariadb",
+        "Percona-Server-client-51",
+        "Percona-Server-client-55"
+      ]
+    },
+    "mysql": {
+      "replace_packages": [
+        "mysql-server",
+        "mariadb-server",
+        "Percona-Server-server-51",
+        "Percona-Server-server-55"
+      ]
+    },
+    "package_replacements": {
+      "mysql-libs": {
+        "enabled": true,
+        "strategy": "yum_shell",
+        "from": "replace_packages",
+        "to": "replace_package_targets",
```

Comment:
most of these settings appear to be boiler plating that could otherwise be convention instead of config.
e.g. yum_shell expects an array to, yum_plugin expects a scaler. These being assumed by default rather than having to explicitly keep defining them. Config being implicitly enabled instead of explicitly (tho can be disabled)